rasa/core/actions/forms.py

```diff
@@ -88,17 +94,51 @@ def from_entity(
             "group": group,
         }
 
+    def _update_not_intent_key_with_global_param(
+        self, form_data: Union[Dict[Text, Any], List[Text]], slot_to_fill: Text
+    ) -> Union[Text, List[Text]]:
+        """Updates the value/s of the `not_intent` key of the slot_to_fill.
+
+        Args:
+            form_data: All the available data (parameters and their values) of the form.
+            slot_to_fill: The slot to be updated.
+
+        Returns:
+            The value/s for the `not_intent` key of the given slot (slot_to_fill)
+            updated with the `global_not_intent` parameter (defined in `domain.yml`).
+        """
+        global_not_intents = form_data.get("global_not_intent")
+        # Gets the form slots data with the use of `required_slots` keyword.
+        if "required_slots" in form_data:
+            form_data = form_data.get("required_slots")
+        # Checks if the global parameter is a list. If not, convert to list.
+        if not isinstance(global_not_intents, list):
+            global_not_intents = [global_not_intents]
+        # Checks that `not_intent` parameter is present in the slot keys.
+        if "not_intent" in form_data[slot_to_fill][0].keys():
+            slot_not_intent = form_data[slot_to_fill][0]["not_intent"]
+            # Checks that the value of `not_intent` is a list.
+            if isinstance(slot_not_intent, list):
+                slot_not_intent.extend(global_not_intents)
+                # Use of set() to remove duplicates.
+                slot_not_intent = list(set(slot_not_intent))
+            elif isinstance(slot_not_intent, str):
+                global_not_intents.append(slot_not_intent)
+                slot_not_intent = list(set(global_not_intents))
+        else:
+            slot_not_intent = global_not_intents
+        return slot_not_intent
+
     def get_mappings_for_slot(
         self, slot_to_fill: Text, domain: Domain
     ) -> List[Dict[Text, Any]]:
         """Get mappings for requested slot.
 
         If None, map requested slot to an entity with the same name
         """
-
         requested_slot_mappings = self._to_list(
             domain.slot_mapping_for_form(self.name()).get(
```

Comment:
I think we can simplify the changes a bit:
1. Adapt `required_slots` as you've done already (no need to consider a missing `required_slot` key as this should've been consolidated by the `Domain` beforehand)
2. Add a new method `get_global_not_intents` which retrieves a domain and returns a list of potentially ignored intents from the domain
3. Make `intent_is_desired` a object method by adding `self` and removing the `staticmethod` decorator
4. make the domain a param of `intent_is_desired`
4. Concatenate `mapping_not_intents` with `self.get_global_not_intents(domain)`

Does that make sense?